encoding/json/inbound.go

```diff
@@ -41,9 +41,12 @@ type jsonHandler struct {
 }
 
 func (h jsonHandler) Handle(ctx context.Context, treq *transport.Request, rw transport.ResponseWriter) error {
+	treq.Encoding = Encoding
+	// TODO(abg): Should we fail requests if Rpc-Encoding does not match?
```

Comment:
👍 
I'm fine with failing on no match. In future, I assume we might have dispatching based on the encoding too (e.g. you have Thrift and Protobuf handlers for the same method, on the same endpoint and service names), so encoding has to be accurate, better to enforce it now.